checker/check_result.go

```diff
@@ -64,11 +100,14 @@ func MakeRetryResult(name string, err error) CheckResult {
 	return CheckResult{
 		Name:        name,
 		Pass:        false,
+		Pass2:       ResultDontKnow,
 		ShouldRetry: true,
 		Error:       scorecarderrors.MakeRetryError(err),
 	}
 }
 
+// TODO: update this function to return a ResultDontKnow
+// if the confidence is low?
```

Comment:
should this decision be left to the check itself, rather than trying to decide for all checks here? Since checks are very different, I think it's hard to do that in this function in practice.

Reply:
We can pass 2 extra args here `resultWhenBelowThreshold` and `resultWhenAboveThreshold`.